examples/next-openai-pages/pages/chat/stream-chat-files/index.tsx

```diff
@@ -0,0 +1,109 @@
+            <div className="w-24 text-zinc-500 flex-shrink-0">{`${message.role}: `}</div>
+            {typeof message.content === 'string' ? (
+              <div className="w-full">{message.content}</div>
+            ) : (
+              <div className="flex flex-col gap-2">
+                {message.content
+                  ?.filter(content => content.type === 'text')
+                  .map((content, index) => {
+                    if (index === 0) {
+                      return (
+                        <div key={`${message.id}-${index}`} className="w-full">
+                          {content.type === 'text' ? content.text : ''}
+                        </div>
+                      );
+                    } else {
+                      return (
+                        <div
+                          key={`${message.id}-${index}`}
+                          className="w-32 text-xs border h-24 text-ellipsis overflow-hidden rounded-md p-2"
+                        >
+                          {content.type === 'text' ? content.text : ''}
+                        </div>
+                      );
+                    }
+                  })}
+
+                <div className="flex flex-row gap-2">
+                  {message.content
+                    ?.filter(content => content.type === 'image')
+                    .map((content, index) => (
+                      <img
+                        key={`${message.id}-${index}`}
+                        className="w-24 rounded-md"
+                        src={`${content.type === 'image' ? content.image : ''}`}
+                        alt="image"
+                      />
+                    ))}
+                </div>
+              </div>
+            )}
+          </div>
+        ))}
+      </div>
+
+      <form
+        onSubmit={handleSubmit}
+        className="flex flex-col gap-2 fixed bottom-0 p-2 w-full"
+      >
+        <div className="flex flex-row gap-2 fixed right-2 bottom-14 items-end">
+          {Array.from(files).map(file => {
+            const { type } = file;
+
+            if (type.startsWith('image/')) {
+              return (
+                <div key={file.name}>
+                  <img
+                    className="w-24 rounded-md"
+                    src={URL.createObjectURL(file)}
+                    alt="image"
+                  />
+                  <span className="text-sm text-zinc-500">{file.name}</span>
+                </div>
+              );
+            } else if (type.startsWith('text/')) {
+              return (
+                <div
+                  key={file.name}
+                  className="w-24 text-zinc-500 flex-shrink-0 text-sm flex flex-col gap-1"
+                >
+                  <div className="w-16 h-20 bg-zinc-100 rounded-md" />
+                  {file.name}
+                </div>
+              );
+            }
+          })}
+        </div>
+        <input type="file" onChange={handleFileInputChange} multiple />
```

Comment:
For my understanding: we are limited to a single `file` input, correct? (even if it can support multiple files)

Reply:
this now depends on the user now that the api exposes `files` through `handleSubmit`